pkg/skaffold/runner/generate_pipeline.go

```diff
@@ -0,0 +1,388 @@
+	taskDeploy, err := generateDeployTask(config.Pipeline.Deploy)
+	if err != nil {
+		return errors.Wrap(err, "generating deploy task")
+	}
+	tasks = append(tasks, taskDeploy)
+
+	// Generate pipeline from git resource and tasks
+	pipeline, err := generatePipeline(tasks)
+	if err != nil {
+		return errors.Wrap(err, "generating tekton pipeline")
+	}
+
+	// json.Marshal all pieces of pipeline, then convert all jsons to yamls
+	var jsons [][]byte
+	bGitResource, err := json.Marshal(gitResource)
+	if err != nil {
+		return errors.Wrap(err, "marshaling git resource")
+	}
+	jsons = append(jsons, bGitResource)
+	for _, task := range tasks {
+		bTask, err := json.Marshal(task)
+		if err != nil {
+			return errors.Wrap(err, "marshaling task")
+		}
+		jsons = append(jsons, bTask)
+	}
+	bPipeline, err := json.Marshal(pipeline)
+	if err != nil {
+		return errors.Wrap(err, "marshaling pipeline")
+	}
+	jsons = append(jsons, bPipeline)
+
+	var output bytes.Buffer
+	for _, item := range jsons {
+		itemYaml, err := yaml.JSONToYAML(item)
+		if err != nil {
+			return errors.Wrap(err, "converting jsons to yamls")
+		}
+		output.Write(append(itemYaml, []byte("---\n")...))
+	}
+
+	// write all yaml pieces to output
+	if err := ioutil.WriteFile(fileOut, output.Bytes(), 0755); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func generateGitResource() (*tekton.PipelineResource, error) {
+	// Get git repo url
+	getGitRepo := exec.Command("git", "config", "--get", "remote.origin.url")
+	gitRepo, err := getGitRepo.Output()
+	if err != nil {
+		return nil, errors.Wrap(err, "getting git repo from git config")
+	}
+
+	// Create git resource for pipeline from users current git repo
+	return &tekton.PipelineResource{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PipelineResource",
+			APIVersion: "tekton.dev/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "source-git",
+		},
+		Spec: tekton.PipelineResourceSpec{
+			Type: tekton.PipelineResourceTypeGit,
+			Params: []tekton.ResourceParam{
+				{
+					Name:  "url",
+					Value: string(gitRepo),
+				},
+			},
+		},
+	}, nil
+}
+
+func generateBuildTask(buildConfig latest.BuildConfig) (*tekton.Task, error) {
+	if len(buildConfig.Artifacts) == 0 {
+		return nil, errors.New("No artifacts to build")
+	}
+
+	return &tekton.Task{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Task",
+			APIVersion: "tekton.dev/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "skaffold-build",
+		},
+		Spec: tekton.TaskSpec{
+			Inputs: &tekton.Inputs{
+				Resources: []tekton.TaskResource{
+					{
+						Name: "source",
+						Type: tekton.PipelineResourceTypeGit,
+					},
+				},
+			},
+			Steps: []corev1.Container{
+				{
+					Name:       "run-build",
+					Image:      "gcr.io/k8s-skaffold/skaffold:v0.34.0",
+					WorkingDir: "/workspace/source",
+					Command:    []string{"skaffold"},
+					Args: []string{"build",
+						"--filename", "skaffold.yaml",
+						"--profile", "oncluster",
```

Comment:
are we assuming that the user will have a profile named `oncluster` in their skaffold.yaml?

Reply:
answered my own question by reading the code :sweat_smile:

Reply:
yes, we can make this further more better in a follow up PR and ask the user if there is already one exists from the given list. ? 
esp when its a custom builder, it might run on cluser. we just don't know
or just go through profiles and see if "kaniko" section is defined.